src/prommis/uky/costing/tests/test_ree_costing.py

```diff
@@ -1949,7 +1986,314 @@ def test_REE_watertap_costing(self, model, solver):
+            operators_per_shift=[4, 9, 2, 2, 2, 3],
+            hours_per_shift=hours_per_shift,
+            shifts_per_day=shifts_per_day,
+            operating_days_per_year=operating_days_per_year,
+            pure_product_output_rates=pure_product_output_rates,
+            mixed_product_output_rates=mixed_product_output_rates,
+            mixed_product_sale_price_realization_factor=0.65,  # 65% price realization for mixed products
+            # arguments related to total owners costs
+            land_cost=model.fs.land_cost,
+            resources=resources,
+            rates=rates,
+            prices={
+                "dummy": 1 * getattr(pyunits, "USD_" + CE_index_year) / pyunits.kg,
+            },
+            fixed_OM=True,
+            variable_OM=True,
+            feed_input=model.fs.feed_input,
+            efficiency=0.80,  # power usage efficiency, or fixed motor/distribution efficiency
+            chemicals=["dummy"],
+            waste=[
+                "nonhazardous_solid_waste",
+                "nonhazardous_precipitate_waste",
+                "dust_and_volatiles",
+            ],
+            recovery_rate_per_year=model.fs.recovery_rate_per_year,
+            CE_index_year=CE_index_year,
+        )
+
+        # define reagent fill costs as an other plant cost so framework adds this to TPC calculation
+        model.fs.costing.other_plant_costs.unfix()
+        model.fs.costing.other_plant_costs_rule = pyo.Constraint(
+            expr=(
+                model.fs.costing.other_plant_costs
+                == pyunits.convert(
+                    1218073 * pyunits.USD_2016  # Rougher Solvent Extraction
+                    + 48723 * pyunits.USD_2016  # Cleaner Solvent Extraction
+                    + 182711
+                    * pyunits.USD_2016,  # Solvent Extraction Wash and Saponification
+                    to_units=getattr(pyunits, "MUSD_" + CE_index_year),
+                )
+            )
+        )
+
+        # fix costing vars that shouldn't change
+        model.fs.feed_input.fix()
+        model.fs.feed_grade.fix()
+        model.fs.recovery_rate_per_year.fix()
+        model.fs.reagents.fix()
+        model.fs.solid_waste.fix()
+        model.fs.precipitate.fix()
+        model.fs.dust_and_volatiles.fix()
+        model.fs.power.fix()
+
+        # check model structural diagnostics
+        dt = DiagnosticsToolbox(model=model, variable_bounds_violation_tolerance=1e-4)
+        dt.assert_no_structural_warnings()
+
+        QGESSCostingData.costing_initialization(model.fs.costing)
+        QGESSCostingData.initialize_fixed_OM_costs(model.fs.costing)
+        QGESSCostingData.initialize_variable_OM_costs(model.fs.costing)
+
+        solver = get_solver()
+        results = solver.solve(model, tee=True)
+        assert_optimal_termination(results)
+
+        # check model numerical diagnostics
+        dt = DiagnosticsToolbox(model=model, variable_bounds_violation_tolerance=1e-4)
```

Comment:
Is it necessary to re-define this (`dt` already defined in L2241)?

Reply:
Thanks for noting this, 2253 is redundant. I'll remove it.